crates/core/src/factories/core_fetch/test.rs

```diff
@@ -374,11 +374,46 @@ async fn ops_are_cleared_when_agent_not_in_peer_store() {
     tokio::time::sleep(Duration::from_millis(10)).await;
 
     // Check that all op ids for agent have been removed from ops set.
-    assert!(fetch.state.lock().unwrap().ops.is_empty());
+    assert!(fetch.state.lock().unwrap().requests.is_empty());
+}
+
+#[test]
+fn back_off() {
+    let back_off_interval_ms = 10;
+    let mut back_off_list = BackOffList::new(back_off_interval_ms, 2);
+    let agent_id = random_agent_id();
+    back_off_list.back_off_agent(&agent_id);
+    assert!(back_off_list.is_agent_backing_off(&agent_id));
+
+    std::thread::sleep(Duration::from_millis(back_off_interval_ms + 1));
```

Comment:
Can this be done by manipulating the clock with Tokio?

Reply:
It can't. The function `is_agent_backing_off` would have to use `tokio::time::Instant::now()` too for this to work. `std::time::Instant`s are unaffected by it.